test/test_parser.cpp

```diff
@@ -386,6 +386,7 @@ Query verify_query(test_util::unit_test::TestContext& test_context, TableRef t,
     CHECK_EQUAL(q_count, num_results);
     std::string description = q.get_description();
     // std::cerr << "original: " << query_string << "\tdescribed: " << description << "\n";
+    mapping.set_backlink_class_prefix("");
```

Comment:
I'd prefer not to do this. There appears to be an asymmetry where the serializer unconditionally removes "class_" prefixes, while table names without prefixes are only translated by the parser to add "class_" if that is set as part of the mapping. I think we should consolidate this so that the serializer has a copy of the mapping state so that it can appropriately apply prefixes when applicable. Then we should be able to support "class_class_..." table names by only removing the first prefix and also remain backwards compatible.

Reply:
Mindblowing. But I think I got it right.